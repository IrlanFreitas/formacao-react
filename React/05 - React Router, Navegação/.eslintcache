[{"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\index.js":"1","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\reportWebVitals.js":"2","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\App.jsx":"3","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\routes\\routes.js":"4","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\components\\Header.jsx":"5","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Sobre.jsx":"6","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\NotFound.jsx":"7","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Home.jsx":"8","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\components\\ListPosts.jsx":"9","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\hook\\useData.js":"10","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\services\\api.js":"11","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Post.jsx":"12","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\components\\ListCategories.jsx":"13","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Category.jsx":"14","C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\SubCategories.jsx":"15"},{"size":494,"mtime":1611147773000,"results":"16","hashOfConfig":"17"},{"size":356,"mtime":1611147773000,"results":"18","hashOfConfig":"17"},{"size":448,"mtime":1616118638614,"results":"19","hashOfConfig":"17"},{"size":848,"mtime":1616205078998,"results":"20","hashOfConfig":"17"},{"size":1517,"mtime":1616199868245,"results":"21","hashOfConfig":"17"},{"size":491,"mtime":1611147773000,"results":"22","hashOfConfig":"17"},{"size":487,"mtime":1616114663364,"results":"23","hashOfConfig":"17"},{"size":381,"mtime":1616205099305,"results":"24","hashOfConfig":"17"},{"size":750,"mtime":1616210445656,"results":"25","hashOfConfig":"17"},{"size":494,"mtime":1616210259508,"results":"26","hashOfConfig":"17"},{"size":230,"mtime":1616123904966,"results":"27","hashOfConfig":"17"},{"size":557,"mtime":1616200778220,"results":"28","hashOfConfig":"17"},{"size":757,"mtime":1616210569818,"results":"29","hashOfConfig":"17"},{"size":1840,"mtime":1616211096966,"results":"30","hashOfConfig":"17"},{"size":329,"mtime":1616206032774,"results":"31","hashOfConfig":"17"},{"filePath":"32","messages":"33","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"ziz0r9",{"filePath":"34","messages":"35","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"36","messages":"37","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"38","messages":"39","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"40","messages":"41","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"42","messages":"43","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"44"},{"filePath":"45","messages":"46","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"44"},{"filePath":"47","messages":"48","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"49","messages":"50","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"51","messages":"52","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"53","messages":"54","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"44"},{"filePath":"55","messages":"56","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"44"},{"filePath":"57","messages":"58","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"59","messages":"60","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"61","messages":"62","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\index.js",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\reportWebVitals.js",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\App.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\routes\\routes.js",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\components\\Header.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Sobre.jsx",[],["63","64"],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\NotFound.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Home.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\components\\ListPosts.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\hook\\useData.js",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\services\\api.js",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Post.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\components\\ListCategories.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\Category.jsx",[],"C:\\Users\\irlan.junior\\Documents\\Pessoal\\Estudos\\Formação React\\React\\05 - React Router, Navegação\\src\\pages\\SubCategories.jsx",[],{"ruleId":"65","replacedBy":"66"},{"ruleId":"67","replacedBy":"68"},"no-native-reassign",["69"],"no-negated-in-lhs",["70"],"no-global-assign","no-unsafe-negation"]