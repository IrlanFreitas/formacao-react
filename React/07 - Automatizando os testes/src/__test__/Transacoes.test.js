import React from "react";

import Transacoes from '../transacoes/Transacoes'
import { render } from "@testing-library/react";

describe('Componente de Listagem de Transações', () => {
    it('O snapshot do componente deve permanecer sempre o mesmo', () => {

        const transacoes = [{
            id: 1,
            tipo: 'saque',
            valor: '30.00',
            data: '01/01/2021',
        }, {
            id: 2,
            tipo: 'deposito',
            valor: '450.00',
            data: '15/01/2021',
        }]

        const { container } = render(<Transacoes transacoes={transacoes} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})
