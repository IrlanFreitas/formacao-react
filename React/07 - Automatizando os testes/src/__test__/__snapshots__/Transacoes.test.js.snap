// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Listagem de Transações O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      01/01/2021
    </p>
    <p />
    <p>
      R$ 
      30.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      15/01/2021
    </p>
    <p />
    <p>
      R$ 
      450.00
    </p>
  </div>
</div>
`;
